### ISSUE TITLE
Try numpy's masked arrays for polar view masking

### ISSUE THREAD

**Issue by psavery** [HEXRD/hexrdgui#380]
This might allow us to switch between masked/unmasked views.

**Comment by bnmajor** [comment]
`Numpy` offers the [`numpy.ma.MaskedArray`](https://numpy.org/doc/stable/reference/maskedarray.baseclass.html#the-maskedarray-class) class which allows setting a mask on data and also allows accessing either the original data `MaskedArray.data`) or the masked data (`MaskedArray.mask`). Only one mask can be set on the data at a time, which means that using this class would require creating a master mask which could then be selected or ignored with the `mask` and `data` attributes. In the case of toggling and removing masks for the mask manager discussed in #385 this does not seem particularly useful (toggling individual masks is already being handled without the `MaskedArray` class in #446) but may serve another purpose elsewhere.

**Comment by joelvbernier** [comment]
Yes!  I 100% think we should use a masked array for the polarview as it plays so nicely with the column summing.  Then as the mask array (boolean) is manipulated with the manager, we just update the mask in the ma and the sum with be updated automagically.  I think it will plot as an unmasked image still, but I am not positive abou that (need to test).

**Comment by bnmajor** [comment]
@joelvbernier For mask manipulation and the mask manager using `MaskedArray` would be an additional step, as we would need to keep the current approach we have as well - keeping track of each individual mask and which view(s) they belong to. Are there other needs or use cases that this would serve to simplify or improve?

**Comment by psavery** [comment]
We do use masked arrays and we do currently have mechanisms in place for showing/hiding masks.
